script.go

```diff
@@ -650,6 +651,40 @@ func (p *Pipe) Get(url string) *Pipe {
 	return p.Do(req)
 }
 
+// Hash returns the hex-encoded hash of the entire contents of the
+// pipe based on the provided hasher, or an error.
+// To perform hashing on files, see [Pipe.HashSums].
+func (p *Pipe) Hash(hasher hash.Hash) (string, error) {
+	if p.Error() != nil {
+		return "", p.Error()
+	}
+	_, err := io.Copy(hasher, p)
+	if err != nil {
+		p.SetError(err)
+		return "", err
+	}
+	return hex.EncodeToString(hasher.Sum(nil)), nil
+}
+
+// HashSums reads paths from the pipe, one per line, and produces the
+// hex-encoded hash of each corresponding file based on the provided hasher,
+// one per line. Any files that cannot be opened or read will be ignored.
+// To perform hashing on the contents of the pipe, see [Pipe.Hash].
+func (p *Pipe) HashSums(hasher hash.Hash) *Pipe {
+	return p.FilterScan(func(line string, w io.Writer) {
```

Comment:
Sorry about the last-minute thought, but would it make sense to use `FilterLine' here instead? That's the simple case when there's one line in, one line out. `FilterScan` is more flexible, but you have to print to the writer yourself. I should really have used `FilterLine` to implement `SHA256Sums` in the first place; I'm not sure why I didn't.

Reply:
Maybe it's because `FilterLine` always produces a line, even if it's empty. What do you think makes more sense for files we can't hash? Print an empty line, so the output lines match up with the input, or just produce no output at all?

Reply:
`FilterLine` could work but I'm just wondering if we should make that change because of backwards compatibility purposes? This would be a breaking change for the existing functions - is that okay?

Reply:
@bitfield I made the change. We can always revert it if we decide we don't want to do it.

Reply:
That's a good point about changing the output. I think let's stick with `FilterScan`, then, and if it makes sense to change the behaviour in the future, we'll do it then. "Break one thing at a time" is a good motto!

Reply:
done